pkg/cluster/cluster.go

```diff
@@ -175,14 +175,27 @@ func (p *ProviderBase) InitK3sCluster(cluster *types.Cluster) error {
 
 	p.Logger.Infof("[%s] deploying additional manifests", p.Provider)
 
-	// deploy additional UI manifests.
+	// deploy additional components.
+	enabledPlugins := map[string]bool{}
 	if cluster.UI {
-		if _, err := p.execute(&cluster.MasterNodes[0], []string{fmt.Sprintf(deployUICommand,
-			base64.StdEncoding.EncodeToString([]byte(dashboardTmpl)), common.K3sManifestsDir)}); err != nil {
-			return err
+		enabledPlugins["dashboard"] = true
+	}
+	if cluster.EnabledPlugins != nil && len(cluster.EnabledPlugins) > 0 {
```

Comment:
if cluster.EnabledPlugins != nil {
  for xxx := range cluster.EnabledPlugins {
  }
}